src/test/java/test/inject/InjectBeforeAndAfterMethodsWithTestResultSampleTest.java

```diff
@@ -43,7 +45,9 @@ public void before(Method m, ITestResult r) {
   @AfterMethod
   public void after(Method m, ITestResult r) {
     String name = m.getName();
-    Assert.assertEquals(r, m_testResult);
+    //Dont run assertions based on built in "hashCode" comparison because its no longer the same
+    //object, but its a new ITestResult object but with the "Same" content.
```

Comment:
I'm not sure about that. Instead, TestResult should implement hashcode and equals methods.